src/ParseServer.js

```diff
@@ -64,64 +64,68 @@ class ParseServer {
     const {
       appId = requiredParameter('You must provide an appId!'),
       masterKey = requiredParameter('You must provide a masterKey!'),
-      cloud,
-      security,
       javascriptKey,
       serverURL = requiredParameter('You must provide a serverURL!'),
-      serverStartComplete,
-      schema,
     } = options;
     // Initialize the node client SDK automatically
     Parse.initialize(appId, javascriptKey || 'unused', masterKey);
     Parse.serverURL = serverURL;
 
     const allControllers = controllers.getControllers(options);
-
-    const { loggerController, databaseController, hooksController } = allControllers;
+    options.state = 'initialized';
     this.config = Config.put(Object.assign({}, options, allControllers));
+    logging.setLogger(allControllers.loggerController);
+  }
 
-    logging.setLogger(loggerController);
+  /**
+   * Starts the Parse Server to be served as an express. Resolves when parse-server is ready to accept external traffic.
+   * @returns {Promise<function>} express middleware
+   */
 
-    // Note: Tests will start to fail if any validation happens after this is called.
-    databaseController
-      .performInitialization()
-      .then(() => hooksController.load())
-      .then(async () => {
-        if (schema) {
-          await new DefinedSchemas(schema, this.config).execute();
-        }
-        if (serverStartComplete) {
-          serverStartComplete();
-        }
-      })
-      .catch(error => {
-        if (serverStartComplete) {
-          serverStartComplete(error);
+  async startApp() {
+    try {
+      if (this.config.state === 'ok') {
```

Comment:
As i understand so this fn replace `serverStartComplete` ?

Reply:
Yes, we won't use deprecation for this.

Reply:
What was the reason to remove `serverStartComplete`? Couldn't it still be convenient to have a hook that is called when the server started? It seems we also keep `serverCloseComplete`.

Reply:
I was thinking that if we are changing the startup method to require `await start()` to be called, there is no purpose for `serverStartComplete`. Perhaps the preferred approach is to use the approach in #7914 where `await start` is optional; developers can mount the Parse Server instance before it's started if they wish.

Reply:
> developers can mount the Parse Server instance before it's started if they wish.

What would be the purpose of that?

Reply:
> What would be the purpose of that?

I guess it would be for making the feature not breaking, so code wouldn't have to be updated. But the current startup method can lead to race conditions / cloud code being ignored, so even though we _could_ make this not a breaking change, I think the startup method **needs** to be changed. 

It's the main challenge I'm having in getting the tests to pass, as previously it was possible to start and mount a Parse Server that had a startup error (for example, if a database connection string was invalid).

Reply:
Let's take a shortcut and save some headache if we can. We are allowed to do this once a year with a major release and should take the opportunity. It's OK for this to be breaking, no deprecation, no legacy code we'd need to maintain.